src/en-GB/news/blog/2018/ansible-module-to-create-crush-hierarchy/index.md

```diff
@@ -0,0 +1,65 @@
+
+So for this first post, let me introduce an Ansible I wrote for ceph-ansible: **ceph\_crush**.
+
+## I. Rationale
+
+ceph-ansible is feature-full, but we lack modules. I’ve long thought that everything that can be done via a simple `command` task in Ansible does not deserve a module. I was wrong.
+
+Day 2 operations, as we call them, refers to consuming and giving access to the storage. In the context of Ceph, this means several things:
+
+- RGW Configuration
+    
+    - Users
+    - Buckets
+    - Bucket policies
+    - S3 acls
+- RBD
+    
+    - Create/delete/modify RBD images
+    - Map them if kRBD
+- Mon:
+    
+    - create pools
+    - create user and keys
+
+Of course, all of that can be handled by the main playbook, but people are unlikely going to re-run the entire playbook to do that. What they want is a simple playbook via a simple interface to interact with the cluster. They don’t want to know anything about Ceph and its CLI, this only thing they care about is to finalize the task they were assigned too.
+
+One the idea behind this is to unify the operational experience through a standard interface, which Ansible and language description, YAML.
+
+## II. Ceph CRUSH module
+
+This module, as its name state, allows you to create CRUSH hierarchy. The creation is done by passing to each host of your inventory a dictionary containing a set of keys where each determines a CRUSH bucket location. Here is an inventory example:
+
+<table><tbody><tr><td class="code"><pre><span class="line">ceph-osd-0 osd_crush_location="{ <span class="string">'root'</span>: <span class="string">'mon-roottt'</span>, <span class="string">'rack'</span>: <span class="string">'mon-rackkkk'</span>, <span class="string">'pod'</span>: <span class="string">'monpod'</span>, <span class="string">'host'</span>: <span class="string">'ceph-osd-0'</span> }"</span><br></pre></td></tr></tbody></table>
+
+The module is configured like this:
+
+<table><tbody><tr><td class="code"><pre><span class="line"><span class="attr">- name:</span> configure crush hierarchy</span><br><span class="line"><span class="attr">  ceph_crush:</span></span><br><span class="line"><span class="attr">    cluster:</span> <span class="string">"<span class="template-variable">{{ cluster }}</span>"</span></span><br><span class="line"><span class="attr">    location:</span> <span class="string">"<span class="template-variable">{{ hostvars[item]['osd_crush_location'] }}</span>"</span></span><br><span class="line"><span class="attr">    containerized:</span> <span class="string">"<span class="template-variable">{{ docker_exec_cmd }}</span>"</span></span><br><span class="line"><span class="attr">  with_items:</span> <span class="string">"<span class="template-variable">{{ groups[osd_group_name] }}</span>"</span></span><br></pre></td></tr></tbody></table>
```

Comment:
Double curly brackets are causing issues as this syntax is used with nunjucks. In this instance it's causing the content between the brackets not to appear.

Can they be escaped?

E.g.
`\{\{ content here \}\}`

This file contains multiple instances of double curly brackets.

Reply:
Fixed